utils/systemd/monerod.service

```diff
@@ -1,37 +1,84 @@
+### Sample unit file for running monerod as a systemd service ###
+
+### For more information about the options specified in the [Unit] and [Install] sections,
+### please see the systemd.unit(5) man page.
+
 [Unit]
-Description=Monero Full Node
+
+Description=Monero Daemon
 After=network.target
 
+### If monerod is configured to use Tor via "--tx-proxy" and, optionally, "--anonymous-inbound",
+### and if Tor is installed and running as a systemd service,
+### then replace the above "After=network.target" directive with the following two directives:
+
+# After=network.target tor.service
+# Wants=tor.service
+
+### The following [Install] section allows monerod.service to be enabled via "systemctl enable",
+### which in turn causes monerod.service to be automatically started at system startup.
+
+[Install]
+
+WantedBy=multi-user.target
+
+### For more information about the options specified in the [Service] section,
+### please see the systemd.service(5) and systemd.exec(5) man pages.
+
 [Service]
+
+Type=simple
+
+### The following "ExecStart=" line assumes that:
+### (a) the monerod binary has been copied to the /usr/local/bin directory and is owned by the "monero" system user and group,
+### (b) the /etc/monero directory exists and is owned by the "monero" system user and group, and
+### (c) the monerod.conf configuration file exists, is located in /etc/monero, and is owned by the "monero" system user and group.
+
+ExecStart=/usr/local/bin/monerod --config-file /etc/monero/monerod.conf --non-interactive
+RestartSec=60
+TimeoutStopSec=300
+Restart=on-failure
+
+### For more information about the following security, sandboxing, and system call filtering options,
+### please see the systemd.exec(5) man page.
+
+### The following two options assume that the "monero" system user already exists and belongs to the "monero" system group.
+
 User=monero
 Group=monero
-WorkingDirectory=~
+
+## Security
+
+NoNewPrivileges=true
+
+## Sandboxing
+
+ProtectSystem=full
+ProtectHome=true
+
+### Please see the relevant entries in the systemd.exec(5) man page for more information about the following directory sandboxing options.
+### Also, the following two monerod configuration options must be set in /etc/monero/monerod.conf:
+### data-dir=/var/lib/monero
+### log-file=/var/log/monero/monerod.log
+
+RuntimeDirectory=monero
 StateDirectory=monero
 LogsDirectory=monero
+ConfigurationDirectory=monero
+RuntimeDirectoryMode=0700
+StateDirectoryMode=0700
+LogsDirectoryMode=0700
+ConfigurationDirectoryMode=0700
```

Comment:
I'm not clear why you're adding these more restrictive permissions . Bitcoin's example service adds things like this presumably because the Bitcoin daemon is also a _wallet_. Monero's daemon is _not_ a wallet and thus I'm not clear why there's such a need to restrict permissions beyond the default 755?

Reply:
Why not? It is good security practice to restrict as much as possible without affecting functionality and stability, regardless of whether monerod is a wallet or not.

This proposed monerod.service file configures monerod to be run by the `monero` system user. There is therefore no need for other users both within and without the `monero` system group to have access to those directories. In fact, for even more robust sandboxing, `ConfigurationDirectoryMode=` can (and perhaps should) be set to `0400`, because monerod does not need to write to `/etc/monero`.

Reply:
Fair enough.

Reply:
The config directory mode should most definitely be 0400 (letting the process alter its own service config can simplify certain attacks), but logs should probably be 0755 (letting non-privileged users view logs with mundane information can be helpful and shouldn't pose a risk).

Reply:
Setting the mode to 0400 will not prevent the process from altering its own config. Since it's running as the user that owns the file, it can simply call chmod() and then modify the file. The correct way to do it would be to make it owned by user root, group monero, mode 0640 (or 0440).

Reply:
Also, I would agree that logs should be either world- or at least group-readable.

Reply:
Indeed, and why I raised the question. 0700 prevents a normal user viewing daemon logs.

Reply:
Thus I agree that "logs should be either world- or at least group-readable".

Reply:
> Also, I would agree that logs should be either world- or at least group-readable.

There's `sudo [command]`, or `sudo --user=monero [command]` if the user does not want to view the logs as root, for that.

> Indeed, and why I raised the question. 0700 prevents a normal user viewing daemon logs.

Tor uses this approach (of setting `0700` for all of its directories), and it's emphatically not a wallet.

Reply:
Why should one have to sudo just to view a logfile, which emphatically doesn't contain any sensitive information?

Reply:
Further, many systems don't use sudo. That leaves you having to `su` to view the logfile. For what benefit, exactly?